packages/components/.stylelintrc.js

```diff
@@ -0,0 +1,23 @@
+/** @type {import('stylelint').Config} */
+module.exports = {
+	extends: '../../.stylelintrc.json',
+	rules: {
+		'declaration-property-value-disallowed-list': [
+			{
+				'/.*/': '/--wp-admin-theme-/',
+			},
+			{
+				message:
+					'--wp-admin-theme-* variables do not support component theming. Use Sass variables from packages/components/src/utils/theme-variables.scss instead.',
```

Comment:
What about `--wp-components-color-*` variables?  We've been using them in CSS-in-JS-styled components, for example (where SASS variables are not an option)

Reply:
Good question. We can address this separately and incrementally after #58097 is merged so we have variables built into the `COLORS` object. The linting considerations for `--wp-components-color-*` are much more complicated than `--wp-admin-theme-*`, so I'd like to think through that better. Fortunately it's very easy to notice when `--wp-components-color-*` is used without fallbacks, so direct usage is unlikely to cause visual regressions, just a matter of code DRYness.

Interestingly, I found usages of `--wp-components-color-*` in `dataviews` and `edit-site` 😅 I'll address that separately as well.

Reply:
Makes sense.

For next steps, what we could do, at least at a high level, could be:
- to reword the warning message from "Use Sass variables instead" to "Use Sass variables or the `--wp-components-*` CSS variables (with a fallback) instead"
- add more linting rules:
  - in components package, to make sure that `--wp-components-*` CSS variables are always associated with a fallback value
  - outside of components package, we should probably prohibit the usage of `--wp-components-*` CSS variables altogether


What do you think?

Reply:
> Interestingly, I found usages of --wp-components-color-* in dataviews and edit-site 😅  I'll address that separately as well.

Ouch 😅  Yup, should definitely be addressed. Another confirmation that those linter rules could be beneficial!

Reply:
> For next steps, what we could do, at least at a high level

I _think_ there are no cases (at least in SCSS files) where `--wp-components-*` vars need to be used instead of the Sass vars (which already have the correct fallbacks built in), so it's likely better to enforce the Sass var usage.

Outside the components package, there are possible use cases that are legitimate, but we aren't ready to commit to anything yet so a blanket restriction is probably the way to go for now. (And the existing usages don't seem necessary.)

Reply:
Agreed